Assets/PresetsPerFolder/PresetPostProcessor.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Experimental;
+using UnityEditor.Presets;
+using UnityEngine;
+
+namespace PresetsPerFolder
+{
+    /// <summary>
+    /// This class is applying Presets assets automatically to each assets added to the folder containing the Preset or any subfolder.
+    /// The code is divided in 3 parts that setup importer dependencies to make sure the importer of any asset stay determinist.
+    ///
+    /// OnPreprocessAsset:
+    /// For each imported asset, this method goes from the root folder and down to the asset folder
+    /// and register a CustomDependency to this folder in case a Preset is being added/removed at a later time.
+    /// It then load all Presets from that folder and try to apply them to the asset importer.
+    /// If it is applied, we add a direct dependency to the Preset itself so the asset can be re-imported when the Preset values are changed.
+    /// </summary>
+    public class EnforcePresetPostProcessor : AssetPostprocessor
+    {
+        void OnPreprocessAsset()
+        {
+            // We first need to make sure we don't apply this AssetPostprocessor to assets in packages.
+            // And not a .cs file as we don't want to trigger a code compilation every time we create/remove a new Preset.
+            // Last but not least, we need to make sure Presets don't depend on themselves, or that will end-up in an infinite import loop...
+            // There is probably more exceptions to add here depending on your project.
```

Comment:
Suggestion: "There may be more exceptions to add here depending upon your project."